### ISSUE TITLE
Need proper documentation for Authentication and Authorization

### ISSUE THREAD

**Issue by sayedcsekuet** [PythonNest/PyNest#92]
Would you please provide proper documentation for Authentication and Authorization. How to protect route in controller and so on that we get normally in NestJS

**Comment by itay-dar-lmnd** [comment]
@sayedcsekuet Yes! I will supply this docs. In the meanwhile, if you have ideas for more things to improve, i would be happy to hear

**Comment by sayedcsekuet** [comment]
@itay-dar-lmnd Thanks for your reply, It Would be great if you provide routing related documentation also

**Comment by sayedcsekuet** [comment]
@itay-dar-lmnd Any update from your side.

**Comment by ItayTheDar** [comment]
Resolved by #107, which added first-class route guards and comprehensive [authentication and authorization documentation](https://github.com/PythonNest/PyNest/blob/main/docs/guards.md). The guide covers controller- and route-level protection, API keys, bearer tokens, OAuth2 scopes, role-based guards, OpenAPI integration and the **Authorize** button, composition, and testing.

Closing as completed.
